packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -1549,14 +1549,23 @@ class TextSelectionGestureDetectorBuilder {
             case PointerDeviceKind.unknown:
               // On macOS/iOS/iPadOS a touch tap places the cursor at the edge
               // of the word.
+              final TextSelection? previousSelection = renderEditable.selection;
               renderEditable.selectWordEdge(cause: SelectionChangedCause.tap);
+              if (previousSelection != renderEditable.selection) {
```

Comment:
`renderEditable.selection` can be `editableText.textEditingValue.selection` I think, and then we can avoid making a change in `renderEditable`.